src/indieweb_utils/feeds/discovery.py

```diff
@@ -17,14 +17,19 @@ class FeedUrl:
     title: str
 
 
-def discover_web_page_feeds(url: str, user_mime_types: Optional[List[str]] = None) -> List[FeedUrl]:
+def discover_web_page_feeds(
+    url: str, user_mime_types: Optional[List[str]] = None, web_page_request: requests.Response = None
```

Comment:
Since the only thing we do with `web_page_request` if it's passed in is to call `web_page_request.text` on it, I think it would be better to rename this to `html: str = ""` so we don't force users of the indieweb utils to use the requests library.

i.e. We shouldn't care _how_ they fetched html content, just that there is some html content. Also setting the default to be an empty string, rather than None allows us to not worry about handling `None`.

Reply:
Great point!

My original thesis was that we need to gather HTTP request headers to be able to do full feed discovery since feeds may be specified in a request header. On further reflection, we can do this with the `url` parameter and a HEAD request.

I propose that if `html` is specified, we use a HEAD request for feed discovery. If `html` is not specified, we use a regular get request as we do right now.